src/spell.t.sol

```diff
@@ -22,36 +22,72 @@ interface AssessorLike {
+        hevm.store(spell.BL1_ROOT_CONTRACT(), keccak256(abi.encode(address(this), uint(0))), bytes32(uint(1)));
+        hevm.store(spell.CF4_ROOT_CONTRACT(), keccak256(abi.encode(address(this), uint(0))), bytes32(uint(1)));
     }
 
     function testCast() public {
-        address assessor_ = spell.ASSESSOR();
-        AssessorLike assessor = AssessorLike(assessor_);
-        PileLike pile = PileLike(spell.PILE());
-        AuthLike(root_).rely(spell_);
+        AssessorLike br3_assessor = AssessorLike(spell.BR3_ASSESSOR());
+        AssessorLike htc2_assessor = AssessorLike(spell.HTC2_ASSESSOR());
+        AssessorLike ff1_assessor = AssessorLike(spell.FF1_ASSESSOR());
+        NAVFeedLike br3_feed = NAVFeedLike(spell.BR3_FEED());
+        PileLike htc2_pile = PileLike(spell.HTC2_PILE());
+        PileLike ff1_pile = PileLike(spell.FF1_PILE());
+        CoordinatorLike bl1_coordinator = CoordinatorLike(spell.BL1_COORDINATOR());
+        CoordinatorLike cf4_coordinator = CoordinatorLike(spell.CF4_COORDINATOR());
+
+        AuthLike(spell.BR3_ROOT_CONTRACT()).rely(spell_);
+        AuthLike(spell.HTC2_ROOT_CONTRACT()).rely(spell_);
+        AuthLike(spell.FF1_ROOT_CONTRACT()).rely(spell_);
+        AuthLike(spell.DBF1_ROOT_CONTRACT()).rely(spell_);
+        AuthLike(spell.BL1_ROOT_CONTRACT()).rely(spell_);
+        AuthLike(spell.CF4_ROOT_CONTRACT()).rely(spell_);
 
         spell.cast();
 
-        assertEq(assessor.seniorInterestRate(), spell.seniorInterestRate());
-        
-        // TODO: add a few risk group checks here
-        (,,uint ratePerSecond41,,) = pile.rates(41);
-        assertEq(ratePerSecond41, uint(1000000001090820000000000000));
+        assertEq(br3_assessor.seniorInterestRate(), spell.br3_seniorInterestRate());
+        assertEq(htc2_assessor.seniorInterestRate(), spell.htc2_seniorInterestRate());
+        assertEq(ff1_assessor.seniorInterestRate(), spell.ff1_seniorInterestRate());
+
+        assertEq(br3_feed.discountRate(), spell.br3_discountRate());
+
+        (,,uint ratePerSecondHtc2_1,,) = htc2_pile.rates(1);
+        assertEq(ratePerSecondHtc2_1, uint(1000000001585490000));
+        (,,uint ratePerSecondHtc2_16,,) = htc2_pile.rates(16);
+        assertEq(ratePerSecondHtc2_16, uint(1000000002774610000));
+        (,,uint ratePerSecondHtc2_33,,) = htc2_pile.rates(33);
+        assertEq(ratePerSecondHtc2_33, uint(1000000004122270000));
+
+        (,,uint ratePerSecondFf1_1,,) = ff1_pile.rates(1);
+        assertEq(ratePerSecondFf1_1, uint(1000000001696470000));
+        (,,uint ratePerSecondFf1_3,,) = ff1_pile.rates(3);
+        assertEq(ratePerSecondFf1_3, uint(1000000001997720000));
+        (,,uint ratePerSecondFf1_5,,) = ff1_pile.rates(5);
+        assertEq(ratePerSecondFf1_5, uint(1000000004433030000));
+
+        assertEq(bl1_coordinator.minimumEpochTime(), 1 days - 10 minutes);
+        assertEq(cf4_coordinator.challengeTime(), 30 minutes);
```

Comment:
would be cool to also create vars vor Bl1minEpochTime & CF4chellengeTime in the spell